test/unit/spec/_service_spec.rb

```diff
@@ -5,15 +5,15 @@
   before { stub_resources }
 
   let(:chef_run) { ChefSpec::SoloRunner.new(platform: 'ubuntu', version: '14.04').converge(described_recipe) }
-  let(:template) { chef_run.template('/etc/init/kibana.conf') }
```

Comment:
Unit tests with Chefspec are not integration tests - so they don't run a 'converge' outside of the unit test.  If you want to set a node variable you have to set it within the unit test itself - or the unit test would fail (it has not outside context of where to get the node['kibana']['template_file'] attribute from.